tests/sdk/nodejs/provider-outputs/step1/index.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as k8s from "@pulumi/kubernetes";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+
+const ns1 = new k8s.core.v1.Namespace("ns1");
+
+// Create a new provider using the current context.
+const k8s1 = new k8s.Provider("k8s1", {
+    kubeconfig: fs.readFileSync("kubeconfig").toString(),
```

Comment:
Any reason to explicitly read the file contents rather than using the path?

Reply:
@lblackstone thanks for asking, I couldn't recall why and so I tried to use a file path and things got interesting. Turns out that the "cluster replacement" logic in the provider wasn't written with kubeconfig paths in mind. I posted a commit ([ref](https://github.com/pulumi/pulumi-kubernetes/pull/2598/commits/b1736dbff03f48b29a04f8a0e4c2278b0fcfe2e1)) to address the issue and to add tests.

I chose to simplify the logic in `DiffConfig` to simply leverage the input normalization w.r.t. the `server` property.  Seems to me that the server name is a sufficient differentiator, there's no need to consider the other server properties (endpoint, certificates).  Let me know what you think.

Reply:
That sounds reasonable. It's been a long time since I looked at this logic, but I can't recall a specific reason those fields needed to be checked.